### PR TITLE
beps: add more todos and future considerations to bep

### PR DESCRIPTION
**Pull request by drodil**
## Hey, I just made a Pull Request!

these are also as task list in the #22213 but adding them here as well for bookkeeping + some new ideas that should be considered

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: beps/0001-notifications-system/README.md [hunk 1]
```diff
@@ -352,13 +354,15 @@ interface SignalApi {
...
 - Replace absent signal service with long polling. This requires changes to the `signals` plugin as well.
-- Notifications can have severity that is used to determine how notifications are shown to the user
+- Render dynamic values with various different React elements such as the `EntityRefLink` for entity references (for example `{{ user:default/john.doe }}`) in the notification payload
+- Handle `link` values that use route references. For example instead hard-coding link to `/catalog/default/component/artist-web` it should be possible to use `catalogPlugin.routes.catalogEntity` route reference as a link of the notification. This should also allow using parameters to be passed to the route reference
+- Add support for `analyticsApi` to notification actions
+- Add support for user settings to control how notifications are shown to the user and which notifications user wants to receive. This should also include support for different `NotificationProcessor`s that can send notification to external systems
+- Add a sound to be played when notification is received
```
Comment (Rugvip) [inline beps/0001-notifications-system/README.md:364]: Oh dear 😅, but sure, let's investigate

File: beps/0001-notifications-system/README.md [hunk 2]
```diff
@@ -352,13 +354,15 @@ interface SignalApi {
...
+- Add i18n internationalization support for the notification payload
```
Comment (Rugvip) [inline beps/0001-notifications-system/README.md:365]: Possibly weaved into the above work of templating the message. I could also see this being handled by plugin-specific rendering of notifications though.

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

:shipit: after the tweak

File: beps/0001-notifications-system/README.md [hunk 3]
```diff
@@ -352,13 +354,15 @@ interface SignalApi {
...
+#### Future considerations and BEP TODO
 
-- Notification frontend utilizes [Web Notification API](https://developer.mozilla.org/en-US/docs/Web/API/Notification) to notify user for new notifications
-- Unread notifications count is displayed in the title of the page
-- Configuration can be used to enable or disable features in the notification system (Web Notification API, title change, etc.)
 - Replace absent signal service with long polling. This requires changes to the `signals` plugin as well.
-- Notifications can have severity that is used to determine how notifications are shown to the user
+- Render dynamic values with various different React elements such as the `EntityRefLink` for entity references (for example `{{ user:default/john.doe }}`) in the notification payload
+- Handle `link` values that use route references. For example instead hard-coding link to `/catalog/default/component/artist-web` it should be possible to use `catalogPlugin.routes.catalogEntity` route reference as a link of the notification. This should also allow using parameters to be passed to the route reference
+- Add support for `analyticsApi` to notification actions
```
[the remark this answers is not in the document]
Comment (drodil) [reply beps/0001-notifications-system/README.md:362]: We could track the user interactions with notifications like opening the links vs marking done etc.

File: beps/0001-notifications-system/README.md (hunk 1 above)
Comment (drodil) [reply beps/0001-notifications-system/README.md:364]: Oh yeah, sounds should be your core-business 😆

File: beps/0001-notifications-system/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (drodil) [reply beps/0001-notifications-system/README.md:362]: We could track the user interactions with notifications like opening the links vs marking done etc.

[the remark this answers is not in the document]
Comment (Rugvip) [reply beps/0001-notifications-system/README.md:362]: Yep yep, thinking we should put that in the doc

**Pull request merged**